src/TableHeaderCells.js

```diff
@@ -0,0 +1,79 @@
+import React from 'react'
+import PropTypes from 'prop-types'
+import { Table } from 'evergreen-ui'
+import styled from 'styled-components'
+
+const TableSearchHeaderCellDiv = styled.div`
+ display: ruby;
+`
+
+const RenderTableSortHeaderCell = ({columnNumber, headerText, width, orderedColumn, ordering, onClick}) =>
+  <Table.TextHeaderCell
+    className={`header${columnNumber}`}
+    flexBasis={width} flexShrink={100} flexGrow={100}
+    onClick={() => onClick(columnNumber)}>
+    <TableSearchHeaderCellDiv>
+      {headerText}
+      { columnNumber===orderedColumn ?
+        ordering ?
+          <i className={`icon icon-common icon-sort-up`} style={{paddingLeft: `1em`}}/>
+          : <i className={`icon icon-common icon-sort-down`} style={{paddingLeft: `1em`}}/>
+        : <i className={`icon icon-common icon-sort`} style={{paddingLeft: `1em`}}/>
+      }
+    </TableSearchHeaderCellDiv>
+  </Table.TextHeaderCell>
+
+const RenderTableSearchHeaderCell = ({columnNumber, headerText, width, searchedColumn, searchQuery, onChange}) =>
+  <Table.SearchHeaderCell
+    className={`searchheader${columnNumber}`}
+    flexBasis={width} flexShrink={100} flexGrow={100}
+    onChange={value => onChange(value, columnNumber)}
+    value={columnNumber===searchedColumn ? searchQuery : ``}
+    placeholder = {`Search by ${headerText} ...`}
+  />
+
+const TableHeaderCells = ({tableHeader, searchedColumn, searchQuery, onClick, onChange, orderedColumn, ordering}) => {
```

Comment:
Is `TableHeaderCells` a function or a stateless component? At first, by just looking at it, I’d say it’s a function, but then I see it has `propTypes` and I’m confused. I think the design and hierarchy of components here could be clearer.